Ical.Net/DataTypes/Duration.cs

```diff
@@ -18,19 +19,47 @@ public struct Duration
     /// <summary>
     /// Initializes a new instance of the <see cref="Duration"/> struct.
     /// </summary>
-    /// <remarks>All non-null arguments must have the same sign.</remarks>
-    /// <exception cref="ArgumentException">Thrown if not all non-null arguments have the same sign.</exception>
+    /// <remarks>
+    /// All non-null arguments must have the same sign (positive or negative).  
+    /// If any argument is zero, it is treated as <see langword="null"/>.  
+    /// If the arguments have mixed signs, an <see cref="ArgumentException"/> is thrown.  
+    /// </remarks>
+    /// <exception cref="ArgumentException">Thrown if not all non-null arguments have the same sign (positive or negative).</exception>
     public Duration(int? weeks = null, int? days = null, int? hours = null, int? minutes = null, int? seconds = null)
     {
-        var sign = GetSign(weeks) ?? GetSign(days) ?? GetSign(hours) ?? GetSign(minutes) ?? GetSign(seconds) ?? 1;
-        if (
-            ((GetSign(weeks) ?? sign) != sign)
-            || ((GetSign(days) ?? sign) != sign)
-            || ((GetSign(hours) ?? sign) != sign)
-            || ((GetSign(minutes) ?? sign) != sign)
-            || ((GetSign(seconds) ?? sign) != sign))
+        weeks = NullIfZero(weeks);
+        days = NullIfZero(days);
+        hours = NullIfZero(hours);
+        minutes = NullIfZero(minutes);
+        seconds = NullIfZero(seconds);
+
+        var signs = new (string Name, int? Sign)[]
+        {
+            (nameof(weeks), GetSign(weeks)),
+            (nameof(days), GetSign(days)),
+            (nameof(hours), GetSign(hours)),
+            (nameof(minutes), GetSign(minutes)),
+            (nameof(seconds), GetSign(seconds))
+        };
+
+        var sign = signs.FirstOrDefault(s => s.Sign.HasValue).Sign ?? 1;
+
+        var invalidArgumentsBuilder = new System.Text.StringBuilder();
```

Comment:
`Duration` has been changed to `struct` not only but also to make it cheap working with it. By creating a StringBuilder and doing Linq just for validating the input we produce quite some overhead. I suggest doing the input validation similarly to how it was done previously and do the string building only in case of an error. Consider moving the string building to a separate method to keep complexity low.

Reply:
Right, that is bloating and completely unnecessary. Reverted and just added the missing `NullIfZero` for all args.